enumeration/diagnostic/diagnostic.go

```diff
@@ -0,0 +1,54 @@
+package diagnostic
+
+import (
+	"github.com/snyk/driftctl/enumeration/alerter"
+	"github.com/snyk/driftctl/enumeration/remote/alerts"
+	"github.com/snyk/driftctl/enumeration/resource"
+)
+
+type Diagnostic interface {
+	Code() string
+	Message() string
+	ResourceType() string
+	Resource() *resource.Resource
+}
+
+type diagnosticImpl struct {
+	alert alerter.Alert
+}
+
+func (d *diagnosticImpl) Code() string {
+	if _, ok := d.alert.(*alerts.RemoteAccessDeniedAlert); ok {
+		return "ACCESS_DENIED"
+	}
+	return "NOT_IMPLEMENTED"
```

Comment:
Is this user-facing? If so, wouldn't something like UNKNOWN_ERROR be more appropriate?

Reply:
Yeah that's user facing here, I didn't want to return an empty value but `UNKNOWN_ERROR` you proposed is probably better 👍🏻 Will change that thanks